File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        
        int result[] = new int[temperatures.length];

        Deque<Integer> stack = new ArrayDeque();


        for (int n = temperatures.length -1; n >= 0; n--){

            while (stack.size() > 0 && temperatures[stack.peek()] <= temperatures[n]){
                stack.pop();
            }
            
            if (stack.size() == 0){
                result[n] = 0;
            }else{
                result[n] = stack.peek() -n;
            }
            stack.push(n);
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        result = ""
        max_length = 0

        for i in range(len(s)):

            l, r = i, i

            while l >= 0 and r < len(s) and s[l]==s[r]:

                if r-l+1 > max_length:
                    result = s[l:r+1]
                    max_length = r - l + 1
                l -= 1
                r += 1 
            
            l, r = i, i +1

            while l >= 0 and r < len(s) and s[l] == s[r]:
                
                if r - l + 1 > max_length:
                    result = s[l:r+1]
                    max_length = r - l
                r += 1
                l-=1

        return result

